helper/schema/field_writer_map.go

```diff
@@ -277,7 +277,7 @@ func (w *MapFieldWriter) setSet(
 		// not the `value` directly is because this forces all types
 		// to become []interface{} (generic) instead of []string, which
 		// most hash functions are expecting.
-		s := &Set{F: schema.Set}
```

Comment:
No worries! I think this may be the first TypeSet using the default implementation that gets written, since the behavior I saw was a nil reference panic, and we definitely would have noticed that elsewhere.